src/main/java/com/answerdigital/answerking/service/CategoryService.java

```diff
@@ -36,84 +37,173 @@ public CategoryService(final ProductService productService,
         this.categoryRepository = categoryRepository;
     }
 
+    /**
+     * Creates a Category.
+     * @param categoryRequest The CategoryRequest.
+     * @return The newly persisted Category, in the form of a CategoryResponse.
+     * @throws NameUnavailableException When the Category name already exists.
+     */
+    @Transactional
     public CategoryResponse addCategory(final CategoryRequest categoryRequest) {
-        if (categoryRepository.existsByName(categoryRequest.name())) {
-            throw new NameUnavailableException(String.format("A category named '%s' already exists", categoryRequest.name()));
-        }
+        validateCategoryNameDoesNotExist(categoryRequest.name(), Optional.empty());
 
-        final Category newCategory = categoryMapper.addRequestToCategory(categoryRequest);
-        final var category =  categoryRepository.save(newCategory);
-        return categoryMapper.convertCategoryEntityToCategoryResponse(category);
+        // Create a new category from the Request and persist the initial category.
+        final Category newCategory = requestToCategory(categoryRequest);
+        final Category category =  categoryRepository.save(newCategory);
```

Comment:
I'd simplify this by just having it be 

`Category category = requestToCategory(categoryRequest);
category =  categoryRepository.save(newCategory);`

Reply:
Resolved